python/enterprise_cluster/azure_class.py

```diff
@@ -0,0 +1,567 @@
+                    "destination_port_range": "*",
+                },
+            ).wait()
+
+            # Generic rule to allow all outbound traffic to anywhere
+            self.network_client.security_rules.begin_create_or_update(
+                resource_group_name,
+                nsg_name,
+                "AllowAllOutboundGeneric",
+                {
+                    "priority": 130,
+                    "protocol": "*",
+                    "access": "Allow",
+                    "direction": "Outbound",
+                    "source_address_prefix": "*",
+                    "destination_address_prefix": "*",
+                    "source_port_range": "*",
+                    "destination_port_range": "*",
+                },
+            ).wait()
+
+            self.logger.info("NSG rules configured successfully.")
+        except Exception as e:
+            self.logger.error(f"Failed to create NSG rules: {e}")
+            # TODO(pratik): Remember to raise for each expception written to terminate the process there itself and clean
+            raise
+
+        # Update network interfaces to include the NSG
+        try:
+            nic_names = ["NodeHeadNic"] + [
+                f"Node{i}Nic" for i in range(1, self.config["vm_setup"]["vm_count"])
+            ]
+            for nic_name in nic_names:
+                nic = self.network_client.network_interfaces.get(
+                    resource_group_name, nic_name
+                )
+                nic.network_security_group = {"id": nsg_result.id}
+                self.network_client.network_interfaces.begin_create_or_update(
+                    resource_group_name, nic_name, nic
+                ).wait()
+            self.logger.info("Network interfaces updated with NSG successfully.")
+        except Exception as e:
+            self.logger.error(f"Failed to update network interfaces with NSG: {e}")
+            raise
+
+        self.logger.info("NSG creation and configuration completed.")
+
+    def generate_config_json(self):
+        self.logger.info("Generating configuration JSON file.")
+
+        # Define default config structure with placeholder values
+        config_data = {
+            "nodes": [
+                {
+                    "private_ip": "",
+                    "web_ingress": {
+                        "public_ip": "",
+                        "run_jobs": True,
+                        "ssh_username": self.config["vm_setup"]["ssh_username"],
+                    },
+                    "sql_server": {
+                        "database_dir": "/opt/neuraldb_enterprise/database",
+                        "database_password": "password",
+                    },
+                    "shared_file_system": {
+                        "create_nfs_server": True,
+                        "shared_dir": "/opt/neuraldb_enterprise/model_bazaar",
+                    },
+                    "nomad_server": False,
```

Comment:
Same here we dont need this conf for sql server for all the nodes, rather just on the head node.